3p/integration.js

```diff
@@ -375,21 +387,22 @@ const defaultAllowedTypesInCustomFrame = [
 
 
 /**
- * Gets data encoded in iframe name attribute.
- * @return {!JsonObject}
+ * Initialize 3p frame.
+ * @param {!Window} win
  */
-function getData(iframeName) {
-  try {
-    // TODO(bradfrizzell@): Change the data structure of the attributes
-    //    to make it less terrible.
-    return parseJson(iframeName)['attributes'];
-  } catch (err) {
-    if (!getMode().test) {
-      dev().info(
-          'INTEGRATION', 'Could not parse context from:', iframeName);
-    }
-    return FALLBACK_CONTEXT_DATA;
-  }
+function init(win) {
+  const data = getData();
```

Comment:
OK, just make sure we're on the same page:
eventually we will have the `name` attribute parsed into a module level variable, this needs to done early because the `name` attribute can soon be overwritten by master selection code.

We can all the parse data `config`. And the config is used for 2 separate things:
- window.AMP_MODE
- window.context

I think it's relevant to this PR because I don't see the needs for the `once()` function in the final code.